packages/crypto/src/lib/crypto.ts

```diff
@@ -359,14 +361,45 @@ function base64ToBufferAsync(base64) {
     });
 }
 
-async function getAmdCert(url: string) {
-  // unfortunately, until AMD enables CORS, we have to use a proxy when in the browser
-  // This project is hosted on heroku and uses this codebase: https://github.com/LIT-Protocol/cors-proxy-amd
-  if (isBrowser()) {
-    // CORS proxy url
-    url = `https://cors.litgateway.com/${url}`;
+/**
+ * Asynchronously fetches an AMD certification from a specified URL using a CORS proxy.
+ * This approach is used to bypass CORS restrictions when accessing external APIs that do not support CORS.
+ * It attempts to fetch the AMD cert through a proxy and if the proxy fetch fails, it throws an error.
+ *
+ * @param url The URL from which to fetch the AMD cert.
+ * @returns A Promise that resolves to a Uint8Array containing the AMD certification data.
+ * @throws An error detailing HTTP or network issues encountered during the fetch process.
+ */
+async function getAmdCert(url: string): Promise<Uint8Array> {
+  const proxyUrl = `${LIT_CORS_PROXY}/${url}`;
+
+  log(
+    `[getAmdCert] Fetching AMD cert using proxy URL ${proxyUrl} due to CORS restrictions.`
+  );
+
+  let response;
+
+  try {
+    response = await fetch(proxyUrl);
+    if (!response.ok) {
+      throw new Error(`[getAmdCert] HTTP error! status: ${response.status}`);
+    }
+  } catch (e) {
+    log(`[getAmdCert] Failed to fetch AMD cert from proxy:`, e);
+
+    // Try direct fetch only if proxy fails
+    log('[getAmdCert] Attempting to fetch directly without proxy.');
+    try {
+      response = await fetch(url);
+      if (!response.ok) {
+        throw new Error(`HTTP error! status: ${response.status}`);
+      }
+    } catch (e) {
+      log('[getAmdCert] Direct fetch also failed:', e);
+      throw e; // Re-throw to signal that both methods failed
+    }
   }
```

Comment:
Would be nice to unravel the nested try/catch and local variable mutation here so that the code flow is linear and we don't duplicate error handling. Maybe something like... (caution: I did not run this code, `fetchAsUint8Array()` should be in module scope, and I couldn't include deleted lines in the suggestion so the existing arrayBuffer handling would need to be manually deleted to apply this code change)
```suggestion
  async function fetchAsUint8Array(targetUrl) {
    const res = await fetch(targetUrl);
    if (!res.ok) {
      throw new Error(`[getAmdCert] HTTP error! status: ${response.status}`);
    }
    const arrayBuffer = await res.arrayBuffer();
    return new Uint8Array(arrayBuffer);
  }

   try {
    return await fetchAsUint8Array(proxyUrl);
  } catch (e) {
      log(`[getAmdCert] Failed to fetch AMD cert from proxy:`, e);
  }
  
  // Try direct fetch only if proxy fails
  log('[getAmdCert] Attempting to fetch directly without proxy.');
  try {
    return await fetchAsUint8Array(url);
  } catch (e) {
    log('[getAmdCert] Direct fetch also failed:', e);
    throw e; // Re-throw to signal that both methods failed
  }

```

This structure also has the added bonus that if building the array buffer or converting it to a Uint8Array throws an error during the first attempt against the proxy, we will still try to fetch directly from the AMD service instead of throwing the error to the caller.